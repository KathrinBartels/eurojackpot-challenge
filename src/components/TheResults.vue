<script setup lang="ts">
defineProps<{
  draw: any
}>()

const formatCurrency = (amount: number) => {
  return new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD'
  }).format(amount)
}

const formatDate = (date: string) => {
  return new Intl.DateTimeFormat('en-US', {
    year: 'numeric',
    month: 'long',
    day: 'numeric'
  }).format(new Date(date))
}
</script>

<template>
  <div class="flex flex-col gap-4 list min-w-100" v-if="draw && draw.draws">
    <div v-for="(draw, index) in draw.draws" :key="index" class="flex flex-col gap-4 result border-solid border-bottom border-1 rounded p-6 border-gray-300 bg-white text-gray-800 shadow-md
    ">
      <h2 class="mt-2">
        {{ formatDate(draw.date) }}
      </h2>

      <div class="flex flex-col gap-2">
        <div class="flex flex-row gap-2">
          <div>Winning numbers:</div>
          <div v-for="(number, index) in draw.numbers" :key="index">
            <span>{{ number }}</span>
          </div>
        </div>

        <div class="flex flex-row gap-2">
          <div>Euro numbers:</div>
          <div v-for="(number, index) in draw.additionalNumbers" :key="index">
            <span>{{ number }}</span>
          </div>
        </div>
      </div>

      <div>
        <span class="font-bold">Jackpot:</span>
        {{ formatCurrency(draw.jackpot) }}
      </div>
    </div>
  </div>
</template>
