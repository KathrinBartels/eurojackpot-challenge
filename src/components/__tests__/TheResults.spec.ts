import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'

import TheResults from '../TheResults.vue'

describe('TheResults', () => {
  it('renders the component with data', async () => {
    const wrapper = mount(TheResults, {
      data() {
        return {
          draw: {
            draw: {
              backendError: null,
              success: true,
              draws: [
                {
                  additionalNumbers: [1, 2],
                  date: '2021-01-01',
                  gameAmount: 1,
                  jackpot: 1,
                  numbers: [1, 2, 3, 4, 5, 6, 7]
                }
              ]
            }
          }
        }
      }
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
