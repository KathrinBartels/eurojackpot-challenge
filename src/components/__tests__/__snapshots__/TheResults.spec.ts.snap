// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TheResults > renders the component with data 1`] = `"<!--v-if-->"`;
